src/plugins/intel_gna/src/frontend/model_quantizer.hpp

```diff
@@ -10,119 +10,83 @@
 
 /**
  * Quantize entire network
  */
 class ModelQuantizer {
- public:
-    // Below four functions are used only in unit tests
-    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork& model, float scale_factor) const {
-        return quantize(model, [](const InferenceEngine::CNNNetwork&, bool run_before_copy, bool low_precision){}, scale_factor);
-    }
-
-    template <class PreQuantisationCb>
-    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork& model, const PreQuantisationCb& cb, float scale_factor) const {
-        return quantize(model, cb, std::vector<float>({scale_factor}));
-    }
-
-    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork& model, std::vector<float> scale_factors) const {
-        return quantize(model, [](InferenceEngine::CNNNetwork&, bool run_before_copy, bool low_precision){}, scale_factors);
-    }
-
-    template <class PreQuantisationCb>
-    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork& model,  const PreQuantisationCb& cb, std::vector<float> scale_factors) const {
-        GNAPluginNS::GnaInputs inputs;
-        InferenceEngine::InputsDataMap inputs_map = model.getInputsInfo();
-        int sf_id = 0;
-        for (auto &&input_data : inputs_map) {
-            auto input_layer = getCreatorLayer(input_data.second->getInputData()).lock();
-            if (scale_factors.size() <= sf_id) {
-                THROW_GNA_EXCEPTION << "Scale factors are not set for some of the inputs";
-            }
-            inputs[input_layer->name].scale_factor = scale_factors[sf_id++];
-        }
-
-        return quantize(model, cb, inputs, InferenceEngine::Precision::I16, false);
-    }
+    const Config& gna_config;
+    const bool fake_quantized;
 
-    // This function is acutally used by the plugin
+public:
+    ModelQuantizer(const Config& gna_config, const bool fake_quantized)
+        : gna_config(gna_config),
+          fake_quantized(fake_quantized) {}
     template <class PreQuantisationCb>
     InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork& model,
                                          const PreQuantisationCb& cb,
-                                         const GNAPluginNS::GnaInputs& inputs,
-                                         const InferenceEngine::Precision& gna_precision,
-                                         const bool& inputs_int8_precision) const {
+                                         const GnaInputs& inputs) const {
         OV_ITT_SCOPED_TASK(itt::domains::GNA_LT, "ModelQuantizer::quantize");
-        auto visitor = [&](InferenceEngine::CNNLayerPtr lp) {
-            auto new_layer = InferenceEngine::injectData<QuantizedLayerParams>(lp);
-            transformLayer(new_layer, WeightsConverter());
+        auto visitor = [&](InferenceEngine::CNNLayerPtr layer_ptr) {
+            auto new_layer = InferenceEngine::injectData<frontend::QuantizedLayerParams>(layer_ptr);
```

Comment:
is this really needed? `frontend::`  QuantizedLayerParams is part of the same namespace as the ModelQuantizer class.